images/devtools-terraform-v1beta1/context/Dockerfile

```diff
@@ -51,6 +52,10 @@ COPY --from=docker-lock /prod/docker-lock /usr/local/bin/docker-lock
 
 COPY terraform-switcher_${tfswitcher_version}_checksums.txt /var/tmp/terraform-switcher_${tfswitcher_version}_checksums.txt
 
+# Setting TF_PLUGIN_CACHE_DIR makes things a lot faster but this somehow
+# causes weird permission issues.
+# ENV TF_PLUGIN_CACHE_DIR=/root/.cache/terraform.d/plugin-cache
```

Comment:
Shouldn't the Docker image supply a non-root user? (And switch this this user at some point)

Reply:
These images only work correctly when running with rootless docker, as they have to write output as the user.

When running with rootless docker there is no persistence of ownership, and ownership of files that are mounted into the container cannot be differentiated. 

If this runs as non root user inside the container things work fine if the image is operating on a pristine repo, but as soon as the repo contains some file that has to be replaced or modified things fall apart as the files are not owned by the non root user.

I will make a draft PR on this branch to try and illustrate this problem and also add some notes to the Readme regarding This.

Reply:
see https://github.com/coopnorge/engineering-docker-images/pull/62 which illustrates the problem somewhat.

I have also added some additional info to the README to explain the situation. But in brief, running as root user when your docker is rootless is really just running as your user, the docker prohibition against root stems from a situation that arises due to docker's security model and the daemon also running as root.